api/src/components/reference/routes.ts

```diff
@@ -11,15 +11,31 @@ export const create = middy(referenceController.create)
     .use(middleware.doNotWaitForEmptyEventLoop({ runOnError: true, runOnBefore: true, runOnAfter: true }))
     .use(middleware.httpJsonBodyParser())
     .use(middleware.authentication())
+    .use(middleware.checkOwnership())
     .use(middleware.validator(referenceSchema.create, 'body'));
 
 export const update = middy(referenceController.update)
     .use(middleware.doNotWaitForEmptyEventLoop({ runOnError: true, runOnBefore: true, runOnAfter: true }))
     .use(middleware.httpJsonBodyParser())
     .use(middleware.authentication())
+    .use(middleware.checkOwnership())
     .use(middleware.validator(referenceSchema.update, 'body'));
 
+export const updateAll = middy(referenceController.updateAll)
+    .use(middleware.doNotWaitForEmptyEventLoop({ runOnError: true, runOnBefore: true, runOnAfter: true }))
+    .use(middleware.httpJsonBodyParser())
+    .use(middleware.authentication())
+    .use(middleware.checkOwnership())
```

Comment:
I wonder if there's a better way of doing this,  - checkownership also checks the authentication so having both is duplication

looking at checkowner  you can use the user object in the request.event that authentication adds perhaps?